main.py

```diff
@@ -1,146 +1,219 @@
+    mkdir_ignore_file_exist(f"{mission_base_path}/tactical/ScatteredHeat")
+
+    mkdir_ignore_file_exist(f"{mission_base_path}/videos")
+
+    return mission_base_path
+
+
+def create_product_from_file_path(file_path: str) -> Product:
+    """Takes in a file path and returns a Product"""
+
+    product = None
+    last_modified_on = datetime.fromtimestamp(os.path.getmtime(file_path)).astimezone(
+        timezone.utc
+    )
+
+    if "images" in file_path:
+        if "EO" in file_path:
+            product = Product("image", "EO", last_modified_on)
+        if "HS" in file_path:
+            product = Product("image", "HS", last_modified_on)
+        if "IR" in file_path:
+            product = Product("image", "IR", last_modified_on)
+    elif "tactical" in file_path:
+        if "Detection" in file_path:
+            product = Product("tactical", "Detection", last_modified_on)
+        if "DPS" in file_path:
+            product = Product("tactical", "DPS", last_modified_on)
+        if "HeatPerim" in file_path:
+            product = Product("tactical", "HeatPerim", last_modified_on)
+        if "IntenseHeat" in file_path:
+            product = Product("tactical", "IntenseHeat", last_modified_on)
+        if "IsolatedHeat" in file_path:
+            product = Product("tactical", "IsolatedHeat", last_modified_on)
+        if "ScatteredHeat" in file_path:
+            product = Product("tactical", "ScatteredHeat", last_modified_on)
+    elif "videos" in file_path:
+        product = Product("video", None, last_modified_on)
+
+    if product is None:
+        raise ValueError(f"Failed to map product: {os.path.basename(file_path)}")
+
+    return product
+
+
+def get_product_s3_key(mission_name: str, product: Product, file_extension: str) -> str:
+    folder = None
+    product_subtype = None
+
+    if product.type == "image":
+        folder = "IMAGERY"
+
+        if product.subtype == "EO":
+            product_subtype = "EOimage"
+        elif product.subtype == "HS":
+            product_subtype = "HSimage"
+        elif product.subtype == "IR":
+            product_subtype = "IRimage"
+
+    elif product.type == "tactical":
+        folder = "TACTICAL"
+        product_subtype = product.subtype
+    elif product.type == "video":
+        folder = "VIDEO"
+        product_subtype = "Video"
+
+    return f"{folder}/{product.timestamp.strftime('%Y%m%d_%H%M%SZ')}_{mission_name}_{product_subtype}{file_extension}"
+
+
+def main() -> None:
+    """Entry point"""
+
+    # Setup
+    config = ConfigManager("config.json")
+    s3_client = S3FileManager(
+        config.aws_access_key_id, config.aws_secret_access_key, config.bucket
+    )
+
+    mission_name, mission_time = get_mission_details()
+    # Create mission file
+    try:
+        s3_client.upload_empty_file(
+            f"MISSION/{mission_name}_{mission_time.strftime('%Y%m%d_%H%M')}Z.txt"
+        )
+        print(f"Created mission: {mission_name}")
```

Comment:
I've been bitten several times by "comment out to test" before. I'd prefer to see a flag set via config. Here's a scrappy example of what I mean:

[storage-mode-flag.diff.txt](https://github.com/intterra-io/intterra-airborne-dsa/files/12397902/storage-mode-flag.diff.txt)